app/code/Magento/Swatches/Test/Mftf/Test/StorefrontRedirectToFirstPageOnFilteringBySwatchTest.xml

```diff
@@ -0,0 +1,102 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="StorefrontRedirectToFirstPageOnFilteringBySwatchTest">
+        <annotations>
+            <features value="Swatches"/>
+            <stories value="Filter by swatch attribute on plp layered navigation"/>
+            <title value="Customers are redirected to first plp page after filtering by swatch"/>
+            <description value="Customers are redirected to first plp page after filtering by swatch"/>
+            <severity value="MINOR"/>
+            <group value="Swatches"/>
+        </annotations>
+
+        <before>
+            <createData entity="SimpleSubCategory" stepKey="createCategory"/>
+            <createData entity="SimpleProduct" stepKey="createSimpleProduct1">
+                <requiredEntity createDataKey="createCategory"/>
+            </createData>
+            <createData entity="SimpleProduct" stepKey="createSimpleProduct2">
+                <requiredEntity createDataKey="createCategory"/>
+            </createData>
+            <createData entity="SimpleProduct" stepKey="createSimpleProduct3">
+                <requiredEntity createDataKey="createCategory"/>
+            </createData>
+
+            <magentoCLI command="config:set catalog/frontend/grid_per_page 1" stepKey="setOneProductPerPage"/>
+            <magentoCLI command="config:set catalog/frontend/grid_per_page_values 1" stepKey="setGridPerPage"/>
+
+            <actionGroup ref="AdminLoginActionGroup" stepKey="loginAsAdmin"/>
+            <actionGroup ref="AddTextSwatchToProductActionGroup" stepKey="addSwatchAttribute">
+                <argument name="usedInLayeredNavigation" value="1"/>
+            </actionGroup>
+            <magentoCLI command="cache:clean block_html collections config full_page" stepKey="cleanInvalidatedCaches"/>
```

Comment:
magentoCron groups=index should be enough, as it cleans cache selectively for the changed products.